parser/types/field_type.go

```diff
@@ -112,11 +118,11 @@ func (ft *FieldType) DelFlag(flag uint) {
 }
 
 func (ft *FieldType) SetFlen(flen int) {
-	ft.flen = flen
+	ft.flen = mathutil.Min(flen, mysql.MaxDecimalWidth)
 }
 
 func (ft *FieldType) SetDecimal(decimal int) {
-	ft.decimal = decimal
+	ft.decimal = mathutil.Min(decimal, mysql.MaxDecimalScale)
```

Comment:
I have two concerns about these two changes:
- If the type is not decimal, why use decimal scale/precision to restrict it?
- Even if the type is decimal, just adjust `flen` and `decimal` in the `Set` function is lack of context, and may cause some unexpected results. For example, because `flen = integer_length + decimal`, so if `decimal` is changed, the `flen` will be affected, but `flen` can not be updated in `SetDecimal`

Reply:
1. non-decimal type, restricting it does not effct anything?
2. if without this restrict as previous, the problem also exists, it happens on any place, it should be protected by the function itself. with this restrict, it is better to remove some overflow errors.

Reply:
What I want to express in my second point is `flen` and `decimal` are coupled together(by the fact that `flen = integer_length + decimal`), but `setDecimal` and `setFlen` are decoupled, it is kind of dangerous, so if you wants to use a common function to adjust the `flen` and `decimal` for decimal type, you should use a new function like `setFlenAndDecimal`.

Reply:
I got it. but the current codes set flen and decimal separately in everywhere.